version: "3.8"

services:
  web:
    build: .
    container_name: fastapi_app
    command: uvicorn app.main:app --reload --host 0.0.0.0 --port 8000
    ports:
      - "8000:8000"
    volumes:
      - .:/app
    depends_on:
      - db
      - redis
      - celery
    networks:
      - mynetwork

  db:
    image: postgres:13
    container_name: postgres_db
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
      POSTGRES_DB: fastapi_db
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - mynetwork

  redis:
    image: "redis:alpine"
    container_name: redis_cache
    ports:
      - "6379:6379"
    networks:
      - mynetwork

  celery:
    build: .
    command: celery -A app.celery_app worker --loglevel=info
    depends_on:
      - redis
      - db
    networks:
      - mynetwork

  nginx:
    image: nginx:latest
    container_name: nginx_proxy
    ports:
      - "80:80"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
    depends_on:
      - web
    networks:
      - mynetwork

  bot:
    build: .
    container_name: telegram_bot
    command: python app/telegram_bot.py
    volumes:
      - .:/app
    environment:
      - PYTHONPATH=/app
      - DATABASE_URL=postgresql://user:password@db/fastapi_db  # Подключение к базе данных
    depends_on:
      - db
    networks:
      - mynetwork

volumes:
  postgres_data:

networks:
  mynetwork:
